lib/event_dispatcher/base_event.rb

```diff
@@ -0,0 +1,21 @@
+module EventDispatcher
+  class BaseEvent
+    private
+
+    attr_reader :propagation_stopped
+
+    public
+
+    def initialize
+      @propagation_stopped = false
+    end
```

Comment:
Minor ruby detail on this one: in ruby you don't actually have to call the `initialize` when extending a class. This means that people may and do forget calling it. So if you're designing classes to be extended, it's a good idea to avoid doing things in the initializer.

In this case, instead of `#initialize` you can just do a

```ruby
private 

def propagation_stopped
  @propagation_stopped ||= false
end
```

This way you'll always get a correctly-initialized value, and you avoid depending on `#initialize`.